runtime/sema/check_variable_declaration.go

```diff
@@ -233,41 +231,6 @@ func (checker *Checker) recordVariableDeclarationRange(
 	)
 }
 
-func (checker *Checker) checkVariableDeclarationUsability(declaration *ast.VariableDeclaration) {
```

Comment:
This check now happens at the expression level for arrays/dictionaries. Because:
- we need to infer the type even in non-variable-delectation scenarios. e.g: the expression `[].getType()` should have a type inferred for the empty array.
- Even if the array is not empty, there could be non-inferrable scenarios
  e.g: `[<some_struct>, <some_resource>]`

Reply:
Nice!